web-api/src/middleware/request_context.rs

```diff
@@ -162,20 +134,36 @@ where
 const TENANT_ID_HEADER: &str = "X-Tenant-Id";
 
 fn extract_tenant_id(
-    config: &tenants::Config,
+    legacy_tenant: Option<TenantId>,
     request: &ServiceRequest,
 ) -> Result<TenantId, anyhow::Error> {
     let header_value = request
         .headers()
         .get(TENANT_ID_HEADER)
-        .map(TenantId::try_from)
+        .map(|header| {
+            //FIXME use <&[u8]>::trim_ascii() once stabilized
+            //  https://github.com/rust-lang/rust/issues/94035
+            TenantId::try_parse_ascii(trim_ascii(header.as_bytes()))
+        })
         .transpose()?;
 
-    match header_value {
+    match (header_value, legacy_tenant) {
         //FIXME in follow up PR this ID will be fetched from the database
         //      during startup/storage initialization.
-        None if config.enable_legacy_tenant => Ok(TenantId::missing()),
-        None => Err(anyhow!("{TENANT_ID_HEADER} header missing")),
-        Some(passed_value) => Ok(passed_value),
+        (None, Some(legacy_tenant)) => Ok(legacy_tenant),
+        (None, None) => Err(anyhow!("{TENANT_ID_HEADER} header missing")),
+        (Some(passed_value), _) => Ok(passed_value),
     }
 }
+
+fn trim_ascii(ascii: &[u8]) -> &[u8] {
+    let start = ascii
+        .iter()
+        .position(|byte| !byte.is_ascii_whitespace())
+        .unwrap_or_default();
+    let end = ascii
+        .iter()
+        .rposition(|byte| !byte.is_ascii_whitespace())
```

Comment:
🔧 `.rposition()` returns the index, that means if we look for the rightmost byte which is not a whitespace, then we have to include it in the returned slice: `&ascii[start..=end]`, which means we have to update the unwrapping as well. or we have to add 1 to the rposition index.

we probably should add a small test for this function.